frameworks/cassandra/tests/test_zzzrecovery.py

```diff
@@ -58,41 +60,65 @@ def test_node_replace_replaces_node():
+    replace_pod = get_pod_to_replace()
+    assert replace_pod is not None, 'Could not find a node to shut down'
 
-    assert node_ip is not None, 'Could not find a node to shut down'
+    # Instead of partitioning or reconnecting, we shut down the host permanently
+    sdk_cmd.shutdown_agent(replace_pod['host'])
 
-    old_agent = get_pod_agent(pod_name)
-    log.info('pod name = {}, node_ip = {}, agent = {}'.format(pod_name, node_ip, old_agent))
+    sdk_cmd.svc_cli(config.PACKAGE_NAME, config.SERVICE_NAME,
+                    'pod replace {}'.format(replace_pod['name']))
+    sdk_plan.wait_for_kicked_off_recovery(config.SERVICE_NAME)
 
-    # instead of partitioning or reconnecting, we shut down the host permanently
-    sdk_cmd.shutdown_agent(node_ip)
+    # Another dump of current cluster tasks, now that repair has started.
+    sdk_cmd.run_cli('task')
 
-    sdk_cmd.svc_cli(config.PACKAGE_NAME, config.SERVICE_NAME, 'pod replace {}'.format(pod_name))
-    sdk_plan.wait_for_kicked_off_recovery(config.SERVICE_NAME)
     sdk_plan.wait_for_completed_recovery(config.SERVICE_NAME)
-
-    log.info('Checking correct number of tasks are running')
     sdk_tasks.check_running(config.SERVICE_NAME, config.DEFAULT_TASK_COUNT)
 
-    log.info('Checking the replaced pod is on a new agent')
-    new_agent = get_pod_agent(pod_name)
-    assert old_agent != new_agent
+    # One last task dump for good measure.
+    sdk_cmd.run_cli('task')
+
+    new_agent = get_pod_agent(replace_pod['name'])
+    log.info('Checking that the original pod has moved to a new agent:\n'
+             'old_pod={}\nnew_agent={}'.format(replace_pod, new_agent))
+    assert replace_pod['agent'] != new_agent
+
+
+def get_pod_to_replace():
+    '''Avoid also killing the system that the scheduler is on. This is just to speed up testing.
+    In practice, the scheduler would eventually get relaunched on a different node by Marathon and
+    we'd be able to proceed with repairing the service from there. However, it takes 5-20 minutes
+    for Mesos to decide that the agent is dead. This is also why we perform a manual 'ls' check to
+    verify the host is down, rather than waiting for Mesos to tell us.
+    '''
+    scheduler_ip = shakedown.get_service_ips('marathon', config.SERVICE_NAME).pop()
+    log.info('Scheduler IP: {}'.format(scheduler_ip))
+
+    pods = {}
+    for pod_id in range(0, config.DEFAULT_TASK_COUNT):
+        pod_name = 'node-{}'.format(pod_id)
+        pods[pod_name] = {
+            'name': pod_name,
+            'host': get_pod_host(pod_name),
+            'agent': get_pod_agent(pod_name)}
+    log.info('Pods:\n{}'.format(pprint.pformat(pods)))
+
+    replace_pod = None
+    for key, value in pods.items():
+        if value['host'] != scheduler_ip:
+            replace_pod = value
+            log.info('Found pod avoiding scheduler at {}: {}'.format(scheduler_ip, value))
+            break
+    return replace_pod
 
 
 def get_pod_agent(pod_name):
```

Comment:
`get_pod_agent` and `get_pod_host` cannot assume the first task in the list of tasks for a pod is representative of all tasks for the pod.  Instead they should look for the `server` task's information.